ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/CommandSubmitterITSpec.scala

```diff
@@ -0,0 +1,259 @@
+      template = "Foo1",
+      weight = 1,
+      payloadSizeBytes = 100,
+    )
+    val foo2Config = WorkflowConfig.SubmissionConfig.ContractDescription(
+      template = "Foo2",
+      weight = 1,
+      payloadSizeBytes = 100,
+    )
+    val consumingExercisesConfig = ConsumingExercises(
+      probability = 1.0,
+      payloadSizeBytes = 100,
+    )
+    val nonconsumingExercisesConfig = NonconsumingExercises(
+      probability = 2.0,
+      payloadSizeBytes = 100,
+    )
+    val config = WorkflowConfig.SubmissionConfig(
+      numberOfInstances = 10,
+      numberOfObservers = 1,
+      uniqueParties = false,
+      instanceDistribution = List(
+        foo1Config,
+        foo2Config,
+      ),
+      nonconsumingExercises = Some(nonconsumingExercisesConfig),
+      consumingExercises = Some(consumingExercisesConfig),
+    )
+
+    for {
+      ledgerApiServicesF <- LedgerApiServices.forChannel(
+        channel = channel,
+        authorizationHelper = None,
+      )
+      apiServices = ledgerApiServicesF("someUser")
+      tested = CommandSubmitter(
+        names = new Names(),
+        benchtoolUserServices = apiServices,
+        adminServices = apiServices,
+        metricRegistry = new MetricRegistry,
+        metricsManager = NoOpMetricsManager(),
+      )
+      (signatory, observers) <- tested.prepare(config)
+      _ <- tested.submit(
+        config = config,
+        signatory = signatory,
+        observers = observers,
+        maxInFlightCommands = 1,
+        submissionBatchSize = 5,
+      )
+      eventsObserver = EventsObserver(expectedTemplateNames = Set("Foo1", "Foo2"))
+      _ <- apiServices.transactionService.transactionTrees(
+        config = WorkflowConfig.StreamConfig.TransactionTreesStreamConfig(
+          name = "dummy-name",
+          filters = List(
+            WorkflowConfig.StreamConfig.PartyFilter(
+              party = signatory.toString,
+              templates = List.empty,
+            )
+          ),
+          beginOffset = None,
+          endOffset = Some(LedgerOffset().withBoundary(LedgerOffset.LedgerBoundary.LEDGER_END)),
+          objectives = None,
+        ),
+        observer = eventsObserver,
+      )
+      observerResult: ObservedEvents <- eventsObserver.result
+    } yield {
+      observerResult.createEvents.size shouldBe config.numberOfInstances withClue ("number of create events")
+
+      observerResult.avgSizeOfCreateEventPerTemplateName("Foo1") shouldBe roughly(
+        foo1Config.payloadSizeBytes * 2,
+        toleranceMul = 0.5,
+      ) withClue ("payload size of create Foo1")
+      observerResult.avgSizeOfCreateEventPerTemplateName("Foo2") shouldBe roughly(
+        foo2Config.payloadSizeBytes * 2,
+        toleranceMul = 0.5,
+      ) withClue ("payload size of create Foo2")
+      observerResult.avgSizeOfConsumingExercise shouldBe roughly(
+        consumingExercisesConfig.payloadSizeBytes * 2,
+        toleranceMul = 0.5,
```

Comment:
Do you know why do we have such a high variance in the returned payload size?

Reply:
I've just checked it in more detail and for input payload size of 100 bytes the payload sizes read from the db were (on subsequent runs) 196, 190, 194 bytes for default sandbox backend (H2?), and 190, 191, 195 for PG.

I allowed for high variance in this assertion so as not to spend too much time on tuning this, and for this test to continue to pass if the actual payload size remains in the same order of magnitude if anything changes in the system (testing on Oracle, dropping H2?).

If we want to have more detailed test for payload sizes then I'd probably do that in a dedicated test case and against different backends.